### PR TITLE
Stop crying wolf

### PR DESCRIPTION
**Pull request by mccalluc**
Silence some react warnings. We don't see these once it's deployed, and I just haven't taken the time before to click around the site and to find them.

### CODE REVIEW

File: context/app/static/js/shared-styles/sections/SectionButtonRow/SectionButtonRow.jsx [hunk 1]
```diff
@@ -14,17 +13,6 @@ function SectionButtonRow({ leftText, buttons, ...props }) {
...
-  /**
-   Text to be displayed in the left most available space. Usually a BottomAlignedTypography component.
-  */
-  leftText: PropTypes.element.isRequired,
-  /**
-   Button(s) to be displayed in the right most available space.
-  */
-  buttons: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.element), PropTypes.element]).isRequired,
-};
-
```
Comment (john-conroy) [inline context/app/static/js/shared-styles/sections/SectionButtonRow/SectionButtonRow.jsx:27]: These comments are used by Storybook as descriptions, are these prop types causing warnings? I'd prefer to keep them.

**Review by john-conroy — CHANGES_REQUESTED** [review]

File: context/app/static/js/components/Detail/MetadataTable/MetadataTable.jsx [hunk 2]
```diff
@@ -99,12 +99,11 @@ function MetadataTable(props) {
...
-    PropTypes.oneOfType(
-      // Usually a string, but placeholder metadata.metadata is an object.
-      [PropTypes.string, PropTypes.object],
-    ),
-  ).isRequired,
+  // Metadata values are usually strings
+  // ... but placeholder metadata.metadata is an object
+  // ... and for donors, all the values are wrapped in arrays.
+  // eslint-disable-next-line react/forbid-prop-types
+  metadata: PropTypes.object.isRequired,
```
Comment (john-conroy) [inline context/app/static/js/components/Detail/MetadataTable/MetadataTable.jsx:106]: Unsure if these comments will be used by storybook, as a practice we might want to keep non-description comments outside of the object.

File: context/app/static/js/shared-styles/sections/SectionButtonRow/SectionButtonRow.jsx (hunk 1 above)
Comment (mccalluc) [reply context/app/static/js/shared-styles/sections/SectionButtonRow/SectionButtonRow.jsx:27]: There was an proptypes warning in the log, but I'm not remembering where. Simplest just to revert this for now, and get the rest in.

File: context/app/static/js/components/Detail/MetadataTable/MetadataTable.jsx (hunk 2 above)
Comment (mccalluc) [reply context/app/static/js/components/Detail/MetadataTable/MetadataTable.jsx:106]: Sure, I'll move that up.

**Review by john-conroy — APPROVED** [review]

**Pull request merged**